tests/unit/components/dialog/GitLogDialog.spec.js

```diff
@@ -0,0 +1,139 @@
+import { installQuasarPlugin } from '@quasar/quasar-app-extension-testing-unit-jest';
+import { shallowMount } from '@vue/test-utils';
+import { createI18n } from 'vue-i18n';
+import i18nConfiguration from 'src/i18n';
+import GitLogDialog from 'src/components/dialog/GitLogDialog.vue';
+import DialogEvent from 'src/composables/events/DialogEvent';
+import { gitLog } from 'src/composables/Project';
+
+installQuasarPlugin();
+
+jest.mock('src/composables/events/DialogEvent', () => ({
+  subscribe: jest.fn(),
+}));
+
+jest.mock('src/composables/Project', () => ({
+  gitLog: jest.fn((projectId, commitRef, depth) => {
+    const logs = [];
+    if (depth === 26) {
+      logs.push({
+        oid: 'notToSee',
+      });
+    }
+    for (let index = 0; index < 25; index += 1) {
+      logs.push({
+        oid: `log${index}`,
+      });
+    }
+    return Promise.resolve(logs);
+  }),
+}));
+
+describe('Test component: GitLogDialog', () => {
+  let wrapper;
+  let subscribe;
+  let unsubscribe;
+
+  beforeEach(() => {
+    subscribe = jest.fn();
+    unsubscribe = jest.fn();
+
+    DialogEvent.subscribe.mockImplementation(() => {
+      subscribe();
+      return { unsubscribe };
+    });
+
+    wrapper = shallowMount(GitLogDialog, {
+      props: {
+        projectName: 'test',
+      },
+      global: {
+        plugins: [
+          createI18n(i18nConfiguration),
+        ],
+      },
+    });
+  });
+
+  describe('Test functions', () => {
+    describe('Test function: setBranchName', () => {
+      it('should set branch name on valid event type', () => {
+        expect(wrapper.vm.branchName).toBeNull();
+
+        wrapper.vm.setBranchName({ key: 'GitLog', branch: 'test' });
+        expect(wrapper.vm.branchName).toEqual('test');
+      });
+
+      it('should not set branch name on invalid event type', () => {
+        expect(wrapper.vm.branchName).toBeNull();
+
+        wrapper.vm.setBranchName({ key: 'InvalidEvent', branch: 'test' });
+        expect(wrapper.vm.branchName).toBeNull();
+      });
+    });
+
+    describe('Test function: onLoad', () => {
+      it('should be call with depth 25 on empty logs', async () => {
+        expect(wrapper.vm.logItems).toEqual([]);
+
+        const done = jest.fn();
+        await wrapper.vm.onLoad(0, done);
+        expect(gitLog).toBeCalledWith('test', null, 25);
+      });
+
+      it('should be call with depth 25 and last oid on not empty logs', async () => {
```

Comment:
`called...`

`on not empty logs` -> `when logs are not empty` or something else